services/QuillLMS/app/controllers/feedback_history_ratings_controller.rb

```diff
@@ -18,6 +13,28 @@ def create_or_update
     end
   end
 
+  def mass_mark
+    params[:feedback_history_ids].each do |id|
+      rating = create_or_update_feedback_history_rating(id)
+      if rating.valid?
```

Comment:
This pattern allows rating #1 to be saved, and #2 to not be saved. Just want to call that out. I don't consider this high impact given the context.

Reply:
Actually, I think this particular approach could be confusing.  It saves valid rating models until the first invalid one, then stops and returns the errors for only that first invalid item.  So you could fix it and have a second invalid item that would break, while valid items after the invalid item won't be saved.

It may be cleaner to implement the same behavior we use for `batch` creating `FeedbackHistory` records: https://github.com/empirical-org/Empirical-Core/blob/e8016f231bee0ebcf950b4b48d6e3412d1dcbf61/services/QuillLMS/app/controllers/api/v1/feedback_histories_controller.rb#L27  (Basically, `create` or just `save` all of them, and if any are not valid, return an array of the various errors.)

Worth noting that this assumes we want to save what we can rather than not saving any data if any of it is invalid, which would require some different code to implement.